### DISCUSSION TITLE
ESP8266 Ethernet

### DISCUSSION THREAD

**Discussion by Lagges** [lathoub/Arduino-AppleMIDI-Library#140] [Q&A] [answered]
Is it possible to use an Wemos D1 Mini Board with an W5500 Breakout Board? I wasn't able to get it to work, but I think that I'm doing something really wrong. As soon as I copy the Ethernet Examples from the Tutorial I get a lot of weird Erros. I used the examples/AVR_Callbacks/AVR_Callbacks.ino and changed the line `APPLEMIDI_CREATE_DEFAULTSESSION_INSTANCE();` to `APPLEMIDI_CREATE_INSTANCE(EthernetUDP, MIDI ,"AppleMIDI-ESP8266", DEFAULT_CONTROL_PORT )`, because when using the Default Session it tries to use WifiUDP automatically. I don't get what I'm doing wrong.

Hopefully someone can help me.

**Comment by lathoub** [comment, marked as answer]
Hi @Lagges I have an ESP32 working with a W5500 running, over SPI. The Wemos D1 Mini supports SPI, so you should be fine.

Have you tried the [ESP32 with W5500 example](https://github.com/lathoub/Arduino-AppleMIDI-Library/blob/master/examples/ESP32_W5500_Callbacks/ESP32_W5500_Callbacks.ino)?

Here is the create statement for your ESP8266:
https://github.com/lathoub/Arduino-AppleMIDI-Library/blob/ff621429dba8719ce12e353c6e35e609ea84d0e7/examples/ESP32_W5500_Callbacks/ESP32_W5500_Callbacks.ino#L12

**Reply by Lagges** [reply]
I tried this one. The Problem is that I can't get the ETH_Helper.h to work, because the ESP8266 uses a different WiFi Libary. Do I need to read the MAC Address of the Wifi Module to get it to work?

**Reply by lathoub** [reply]
ETH_Helper.h uses the [Ethernet library](https://github.com/arduino-libraries/Ethernet) that you might need to download separately. Use the ESP8266 WiFi library only to get the MAC address (you might want to start with a hardcoded MAC address first, to minimise your code at first)

**Reply by Lagges** [reply]
Got it to work. I used the Ethernet libary that comes with the ESP8266, so no need to install it separately. The Problem I had, was that you have to add a transitor to SCS of the module. That seems to be a problem specific to the EPS8266 as described here https://esp8266hints.wordpress.com/2018/02/13/adding-an-ethernet-port-to-your-esp-revisited/ .
